src/server/WaitBlock.cpp

```diff
@@ -85,11 +72,7 @@ ConsoleWaitBlock::~ConsoleWaitBlock()
 {
     _pProcessQueue->_blocks.erase(_itProcessQueue);
     _pObjectQueue->_blocks.erase(_itObjectQueue);
-
-    if (_pWaiter != nullptr)
-    {
-        delete _pWaiter;
-    }
+    delete _pWaiter;

```

Comment:
Because the constructor threw if null? Yeah?

Reply:
(Might want to write FAIL_FAST_IF_NULL on it in case some future goof violates your assumption.)

Reply:
`delete` is a no-op for `nullptr`! That's why you don't have to check for `nullptr` first. 🙂
That's basically a drive-by fix. I only noticed it while working on the other changes in this file.

Reply:
Oh ho. Sneaky sneaky.